core/chains/evm/client/rpc_client.go

```diff
@@ -0,0 +1,945 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+
+	clienttypes "github.com/smartcontractkit/chainlink/v2/common/chains/client"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type RPCClient interface {
+	clienttypes.ChainRPCClient[
+		*big.Int,
+		common.Hash,
+		*evmtypes.Head,
+		ethereum.Subscription,
+	]
+	clienttypes.RPCClient[
+		*big.Int,
+		evmtypes.Nonce,
+		common.Address,
+		common.Hash,
+		types.Transaction,
+		common.Hash,
+		types.Log,
+		ethereum.FilterQuery,
+		*evmtypes.Receipt,
+		*assets.Wei,
+		*evmtypes.Head,
+		ethereum.Subscription,
+	]
+}
+
+type rpcClient struct {
```

Comment:
From a quick look, it seems like you have duplicate code between the generalized node struct and the rpcClient. There should be some way (like extending or injecting) we can avoid this